src/state.rs

```diff
@@ -93,15 +99,42 @@ impl BarState {
     }
 
     pub(crate) fn set_message(&mut self, now: Instant, msg: Cow<'static, str>) {
-        self.state.message = msg;
+        self.state.message = TabExpandedString::new(msg, self.tab_width);
         self.update_estimate_and_draw(now);
     }
 
+    // Called in builder context
+    pub(crate) fn set_message_without_draw(&mut self, msg: Cow<'static, str>) {
```

Comment:
I don't see much point in having this setter, we could just make the necessary fields `pub(crate)`?

Reply:
I made them setters because they have invariants they need to maintain. Namely, when setting the message and prefix, we need to keep the tab widths of everything in sync.

Reply:
That doesn't seem like a big issue because `self.state.prefix` is a `TabExpandedString`, so the type system will force the caller to use the correct type. Maybe `ProgressState` should have an `expand(&self, &str) -> TabExpandedString` method that callers can use?